### PR TITLE
Addon Settings Dialog UI update

### PR DESCRIPTION
**Pull request by EthanLowenthal**
# Goal
The current dialog for managing addons was difficult to read and extend. This PR refreshes the UI of the addon settings dialog, while maintaing its core structure and functionality.

This PR should be merged before #215 (Export Addons)

# Changes
### Addon search path input
The textbox that held all the addon search paths has been replaced by a list of current addon paths as well as a single text input and a 'browse' button to allow users to select a path from their own filesystem. This UI attempts to remove possible error from leaving a wrong character or accidental newline in the old textbox.

The ability to copy/paste the entire list of addon search paths has been lost.

### Currently loaded addon display
The old tables have been replaced by a tab view, which makes navigating currently loaded extentions much easier. Each tab contains a treeview with the applicable currently loaded addons, as well as the files they were loaded from.

# Screenshot
#### The new addon dialog
Screenshot 2024-05-21 at 1 59 10 PM

### CODE REVIEW

File: LabExT/View/Controls/CustomTable.py [hunk 1]
```diff
@@ -173,21 +177,19 @@ def sortby(tree, col, descending):
...
-            tup = (float('+inf'), item[1])
-            data.append(tuple(tup))
-        else:
-            data.append(item)
+    raw_data = zip(*[(tree.set(child, col), child)
+                for child in tree.get_children('')])
 
+    # separate out the column data so we can verify it is all the same type
+    column_values = next(raw_data)
+    column_keys = next(raw_data)
```
Comment (DJAntivenom) [inline LabExT/View/Controls/CustomTable.py:185]: Couldn't this lead to a `StopIteration` being raised if the tree is empty?

File: LabExT/View/Controls/CustomTable.py [hunk 2]
```diff
@@ -197,3 +199,20 @@ def sortby(tree, col, descending):
     # switch so tree will sort in opposite direction next time it is clicked
     tree.heading(
         col, command=lambda col=col: sortby(tree, col, int(not descending)))
+    
+def is_float(s):
+    """Helper function to check if a string is a float
+    Unfortunately this is the most foolproof method in python
```
Comment (DJAntivenom) [inline LabExT/View/Controls/CustomTable.py:205]: XD

**Review by DJAntivenom — COMMENTED** [review]
Looks much better :)

**Review by marcoep — APPROVED** [review]

**Comment by marcoep** [comment]
@EthanLowenthal - pls fix conflicts and re-request review s.t. I can merge this one.

**Pull request merged**